pkg/scannerv4/mappers/mappers.go

```diff
@@ -891,29 +893,138 @@ func redhatCSAFAdvisories(ctx context.Context, enrichments map[string][]json.Raw
 	return ret, nil
 }
 
-// filterPackages filters out packages from the given map.
-func filterPackages(packages map[string]*claircore.Package, environments map[string][]*claircore.Environment, packageVulns map[string][]string) {
+// filterPackages filters out packages from the given vulnerability report.
+func filterPackages(report *claircore.VulnerabilityReport) {
 	// We only filter out Node.js packages with no known vulnerabilities (if configured to do so) at this time.
 	if !features.ScannerV4PartialNodeJSSupport.Enabled() {
 		return
 	}
-	for pkgID := range packages {
-		envs := environments[pkgID]
+	for pkgID := range report.Packages {
+		envs := report.Environments[pkgID]
 		// This is unexpected, but check here to be safe.
 		if len(envs) == 0 {
 			continue
 		}
 		if srcType, _ := scannerv4.ParsePackageDB(envs[0].PackageDB); srcType != storage.SourceType_NODEJS {
 			continue
 		}
-		if len(packageVulns[pkgID]) == 0 {
-			delete(packages, pkgID)
-			delete(environments, pkgID)
-			delete(packageVulns, pkgID)
+		if len(report.PackageVulnerabilities[pkgID]) == 0 {
+			delete(report.Packages, pkgID)
+			delete(report.Environments, pkgID)
+			delete(report.PackageVulnerabilities, pkgID)
 		}
 	}
 }
 
+// filterVulnerabilities filters out vulnerabilities from the given vulnerability report.
+// Note: This function only modifies the report's PackageVulnerabilities map.
+// It is non-trivial to remove all unreferenced vulnerabilities from the report's
+// Vulnerabilities map, so we leave it untouched and accept we may send Scanner V4 clients
+// extra vulnerabilities. It is up to the client to handle this.
+func filterVulnerabilities(report *claircore.VulnerabilityReport) {
+	// We only filter out non-Red Hat vulnerabilities found in Red Hat layers (if configured to do so) at this time.
+	if !features.ScannerV4RedHatLayers.Enabled() {
+		return
+	}
+
+	redhatLayers := rhccLayers(report)
```

Comment:
Not something very important but the variable name can be more generic, something like `pkgLayersToKeep` since the method name indicates it's not just for filtering non-Redhat layers vulns.

Reply:
since this is currently implemented specifically for Red Hat, I'm ok with keeping it like this instead of more generic. Let me know if you think otherwise